app/experimenter/experiments/tests/api/v5/test_queries.py

```diff
@@ -385,6 +385,70 @@ def test_experiment_in_review_can_review(self):
         experiment_data = content["data"]["experimentBySlug"]
         self.assertTrue(experiment_data["canReview"])
 
+    def test_experiment_no_rejection_data(self):
+        user_email = "user@example.com"
+        experiment = NimbusExperimentFactory.create_with_status(
+            NimbusExperiment.Status.DRAFT,
+            publish_status=NimbusExperiment.PublishStatus.IDLE,
+        )
+
+        response = self.query(
+            """
+            query experimentBySlug($slug: String!) {
+                experimentBySlug(slug: $slug) {
+                    rejection {
+                        changedBy {
+                            email
+                        }
+                    }
+                }
+            }
+            """,
+            variables={"slug": experiment.slug},
+            headers={settings.OPENIDC_EMAIL_HEADER: user_email},
+        )
+        self.assertEqual(response.status_code, 200, response.content)
+        content = json.loads(response.content)
+        experiment_data = content["data"]["experimentBySlug"]
+        self.assertIsNone(experiment_data["rejection"])
+
+    def test_experiment_with_rejection(self):
+        user_email = "user@example.com"
+        experiment = NimbusExperimentFactory.create_with_status(
+            NimbusExperiment.Status.DRAFT,
+            publish_status=NimbusExperiment.PublishStatus.IDLE,
+        )
+
+        for publish_status in (
+            NimbusExperiment.PublishStatus.REVIEW,
+            NimbusExperiment.PublishStatus.IDLE,
+        ):
+            experiment.publish_status = publish_status
+            experiment.save()
+            generate_nimbus_changelog(experiment, experiment.owner)
```

Comment:
I've seen this pattern in a lot of tests. Worth turning it into a helper?

Reply:
Yeah I was thinking that I think when we move everything over to actually using publish_status we'll update the `create_with_status` method on the factory to include moving through the publish_statuses but when I tried now it didn't make sense without all the other bits being hooked up so we can do it after we wrap everything else up as an optimization.  But yes 👍

Reply:
Filed https://github.com/mozilla/experimenter/issues/4902 as a followup